Assets/Scripts/UI/LoadScreenUI.cs

```diff
@@ -0,0 +1,66 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Playables;
+using UnityEngine.SceneManagement;
+using UnityEngine.UI;
+
+public class LoadScreenUI : MonoBehaviour
+{
+    #region Editor Fields
+
+    [SerializeField] private GameObject _loadingPanel;
+    [SerializeField] private Slider _loadingBarSlider;
+
+    #endregion
+
+    #region Private Variables
+
+    private static LoadScreenUI _instance;
+
+    #endregion
+
+    #region Public Properties
+
+    public static LoadScreenUI Instance { get { return _instance; } }
+
+    #endregion
+
+    #region Unity Loops
+
+    private void Awake()
+    {
+        if (_instance != null && _instance != this)
+        {
+            Destroy(this.gameObject);
+        }
+        else
+        {
+            _instance = this;
+        }
+    }
+
+    #endregion
+
+    public void LoadScene(AsyncOperation operation)
+    {
+        StartCoroutine(LoadSceneAsync(operation));
+    }
+
+    private IEnumerator LoadSceneAsync(AsyncOperation operation)
```

Comment:
Why does this need to be an IEnumarator?

Reply:
So it can be done asynchronously